cli/ripsaw/commands/operator.py

```diff
@@ -0,0 +1,114 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""CLI Commands for benchmark operations"""
+
+import os
+import subprocess
+import tempfile
+
+import click
+import git
+from ripsaw.clients.k8s import Cluster
+from ripsaw.util import logging
+
+logger = logging.get_logger(__name__)
+DEFAULT_REPO = "https://github.com/cloud-bulldozer/benchmark-operator"
+DEFAULT_BRANCH = "master"
+
+
+@click.group("operator")
+def operator_group():
+    """Passthrough function for Click CLI group, used as a decorator to declare functions for the CLI"""
+
+
+@operator_group.command("install")
+@click.option("--repo", help="Repo URL for benchmark-operator", show_default=True, default=DEFAULT_REPO)
+@click.option("--branch", help="Branch to checkout", show_default=True, default=DEFAULT_BRANCH)
+@click.option("--kubeconfig", help=" path to kubeconfig of cluster", default=None)
+def install(repo=DEFAULT_REPO, branch=DEFAULT_BRANCH, command="make deploy", kubeconfig=None):
+    """installs the operator from repo into specified cluster"""
+    click.echo(f"Installing Operator from repo {repo} and branch {branch}")
+    _perform_operator_action(repo, branch, command, kubeconfig)
+    wait_for_operator(kubeconfig=kubeconfig)
+    click.secho("Operator Running", fg="green")
+
+
+@operator_group.command("delete")
+@click.option("--repo", help="Repo URL for benchmark-operator", show_default=True, default=DEFAULT_REPO)
+@click.option("--branch", help="Branch to checkout", show_default=True, default=DEFAULT_BRANCH)
+@click.option("--kubeconfig", help="kubeconfig of cluster", default=None)
+def delete(repo=DEFAULT_REPO, branch=DEFAULT_BRANCH, command="make kustomize undeploy", kubeconfig=None):
+    """delete the operator from configured cluster"""
+    click.echo("Deleting Operator")
+    _perform_operator_action(repo, branch, command, kubeconfig)
+    click.secho("Operator Deleted", fg="green")
+
+
+def wait_for_operator(namespace="benchmark-operator", kubeconfig=None):
+    """wait for operator pods to be ready"""
+    cluster = Cluster(kubeconfig_path=kubeconfig)
+    label_selector = "control-plane=controller-manager"
+    cluster.wait_for_pods(label_selector, namespace, timeout=120)
+
+
+def _perform_operator_action(repo, branch, command, kubeconfig=None):
+    """run command from operator repo"""
+    shell_env = os.environ.copy()
+    if kubeconfig is not None:
+        shell_env["KUBECONFIG"] = kubeconfig
+    local_git_repo = _find_git_repo(os.getcwd())
+    result = None
+    if "benchmark-operator" in local_git_repo:
+        shell_env["VERSION"] = "latest"
+        logger.info("Detected CLI is running from local repo, using that instead of remote")
+        result = subprocess.run(
```

Comment:
Can we handle `CalledProcessError` for this?

Reply:
yeah we can, what do you think we should do with it? log/reraise/exit?

Reply:
Whatever you prefer, the stack trace doesn't look like an expected behavior.

Reply:
updated it to make it look cleaner and also refactored it a bit